optaplanner-core/src/main/java/org/optaplanner/core/impl/localsearch/decider/acceptor/greatdeluge/GreatDelugeAcceptor.java

```diff
@@ -0,0 +1,118 @@
+package org.optaplanner.core.impl.localsearch.decider.acceptor.greatdeluge;
+
+import org.optaplanner.core.api.score.Score;
+import org.optaplanner.core.impl.localsearch.decider.acceptor.AbstractAcceptor;
+import org.optaplanner.core.impl.localsearch.scope.LocalSearchMoveScope;
+import org.optaplanner.core.impl.localsearch.scope.LocalSearchPhaseScope;
+import org.optaplanner.core.impl.localsearch.scope.LocalSearchStepScope;
+import org.optaplanner.core.impl.score.ScoreUtils;
+
+public class GreatDelugeAcceptor extends AbstractAcceptor {
+
+    private Score initialLevel = null;
+
+    private Double rainSpeed = null;
+    private Double rainSpeedRatio = DEFAULT_RAIN_SPEED_RATIO;
+
+    private int levelsLength = -1;
+    private double[] initialLevelScoreLevels;
+    private double[] levelScoreLevels;
+
+    private double levelMinimum = 0;
+    private static final double THRESHOLD = .0000001;
```

Comment:
code style: in doubles we put the 0 in front too, so 0.0000001

Reply:
What threshold is this? Can we give it a better self-explaining name?

Reply:
(serious issue) The threshold is to deal with floating point numerical stability.
But why not avoid the entire problem, but never using doubles, but Score instances?
This works:
- score.subtract(rainSpeedScore)
- score.multiply(rainRatioDouble)

ScoreUtils.extractLevelDoubles() is inaccurate. Try a SimpleLongScore of 9007199254740993L. [1], the double will be a different number. Try a SimpleBigDecimalScore of "0.01" with "0.09", similar problem. By keeping them in their original Score type (using getScoreDefinition() methods where needed), that numerical instability goes away.

[1] https://ge0ffrey.github.io/ge0ffrey-presentations/cornerCaseCheatSheet/cheatSheetJava.html

Using

Reply:
This was a good suggestion and I reworked everything according to this.